### ISSUE TITLE
GRU implementation has different behavior with numpy.int64 than LSTM implementation

### ISSUE THREAD

**Issue by boulderZ** [keras-team/keras#10824]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [ x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [ x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

I get an error when running the script below because the GRU version behaves differently with numpy.int64 as input than the LSTM version. This is a problem because you would expect to simply change the implementation from LSTM to GRU without having to change any other inputs to the model. This came up because I was getting some of my parameters from a pandas dataframe that had numpy.int64 dtypes. I recently wanted to try implementing a GRU and when I simply changed my exisiting LSTM implementation to GRU without changing anything else, I got the error from the GRU implementation. The other inputs like input_shape=() does not seem to care about numpy.int64 but the first argument specifying number of units does. Using Keras 2.1.5  and gpu tensorflow 1.7 as that is latest I can install easily. I found same issue running on windows and running on an AWS instance of ubuntu using cpu version of tensorflow. 

```python
from keras.models import Sequential
from keras import layers
import numpy as np

data_dim = 21
timesteps = 8
num_classes = 10
num_neurons = 32
num_neurons = np.int64(num_neurons)

def test_model_GRU(data_dim,timesteps,num_classes):
    # expected input data shape: (batch_size, timesteps, data_dim)
    model = Sequential()
    model.add(layers.GRU(num_neurons,
                         name='gru_1',
                         input_shape=(timesteps, data_dim,),
                         ))
    model.add(layers.Dense(10,
                           activation='softmax',
                           ))
    model.compile(loss='categorical_crossentropy',
                  optimizer='rmsprop',
                  metrics=['accuracy'])
    return model

def test_model_LSTM(data_dim,timesteps,num_classes):
    # expected input data shape: (batch_size, timesteps, data_dim)
    model = Sequential()
    model.add(layers.LSTM(num_neurons,
                         input_shape=(timesteps, data_dim,),
                         ))
    model.add(layers.Dense(10,
                           activation='softmax',
                           ))
    model.compile(loss='categorical_crossentropy',
                  optimizer='rmsprop',
                  metrics=['accuracy'])
    return model

# Generate dummy training data
x_train = np.random.random((1000, timesteps, data_dim))
y_train = np.random.random((1000, num_classes))

# Generate dummy validation data
x_val = np.random.random((100, timesteps, data_dim))
y_val = np.random.random((100, num_classes))


model = test_model_LSTM(data_dim,timesteps,num_classes)
model.fit(x_train, y_train,
          batch_size=64, epochs=5,
          validation_data=(x_val, y_val))

model = test_model_GRU(data_dim,timesteps,num_classes)
model.fit(x_train, y_train,
          batch_size=64, epochs=5,
          validation_data=(x_val, y_val))
```

**Comment by Afrojam** [comment]
I'm getting exactly the same error with Keras 2.2.4:
In LSTM if the units is an int64 there's no error, but in GRU (the normal GRU or the CuDNNGRU) throws an error about no len for int64.
@boulderZ  have you found the reason?
By now I'm just casting to int before calling the GRU function.

**Comment by boulderZ** [comment]
Yeah, I just cast as well. I was just wanting to point out to the developers the issue.

**Comment by Afrojam** [comment]
Got the same issue: no problem with LSTM but problems with GRU.

**Comment by sachinprasadhs** [comment]
@boulderZ , The issue has been resolved in recent Tensorflow version 2.5, I'm able to run the code without any error, please find the gist [here](https://colab.research.google.com/gist/sachinprasadhs/34be15141bc54463a44b61ca312f1d95/untitled.ipynb).
